### PR TITLE
Add support for local plugin.

### PR DESCRIPTION
**Pull request by fennecdjay**
also correct a few coding convention error and trailing spaces

### CODE REVIEW

File: zpm.c [hunk 1]
```diff
@@ -260,18 +275,21 @@ int plugins_update_local_clone() {
...
     printf("Updating plugins ...\n");
     while(plugin_name) {
-      strcpy(command, "cd ~/.zpm/plugins/");
-      strcat(command, plugin_name);
-      strcat(command, "; git pull");
-      printf("Updating %s...\n", plugin_name);
-      ret = system(command);
-      plugin_name = strtok(NULL, "\n");
+        if (plugin_name[0] == '/')
+            goto next;
```
Comment (desyncr) [inline zpm.c]: This 'goto' could be translated to a simple 'if', or is there a reason to use 'goto'?

File: zpm.c [hunk 2]
```diff
@@ -174,13 +187,15 @@ int mkdir_p(const char *path) {
...
         if (errno != EEXIST)
-            return -1; 
+            return -1;
     }
 
     return 0;
 }
 
 int local_clone_exists(char* plugin_name) {
+    if (plugin_name[0] == '/')
```
Comment (desyncr) [inline zpm.c]: Don't forget curly brackets '{}'

**Review by desyncr — CHANGES_REQUESTED** [review]
Few missing things.

File: zpm.c (hunk 1 above)
Comment (fennecdjay) [reply zpm.c]: You're rigth, a 'if' will be better.

File: zpm.c (hunk 2 above)
Comment (fennecdjay) [reply zpm.c]: oups...

**Comment by fennecdjay** [comment]
Sorry for the number of commit. I'm still messing with git I guess.
Anyway the code migth be OK now.

**Comment by desyncr** [comment]
Looking fine. Gonna test this out tomorrow! Have you taken a look at my [proposal](https://github.com/desyncr/zpm-zsh/pull/12#issuecomment-275511197)?

**Pull request merged**